packages/coil-extension/src/background/services/BackgroundScript.ts

```diff
@@ -215,45 +218,50 @@ export class BackgroundScript {
   private setFramesOnChangedListener() {
     // Reset tab state and recheck adapted content when tab changes location
     this.framesService.on('frameChanged', event => {
+      // If not a top level window or its direct descendant, bail early
       if (!(event.frame.top || event.frame.parentFrameId === 0)) {
         return
       }
 
-      const { tabId } = event
-      const status = event.changed.state
+      const { tabId, frameId } = event
+      const changedState = event.changed.state
       const isComplete = event.frame.state === 'complete'
-      const becameComplete = status === 'complete'
+      const becameComplete = changedState === 'complete'
+      const becameLoading = changedState === 'loading'
       const changedUrl = Boolean(event.changed.href)
 
-      // Always get the url from the tab
+      // Always get the current url
```

Comment:
change: always get the url from the full frame, not the bundle of "changed" properties, which may not include the url